pendulum/date.py

```diff
@@ -83,8 +84,11 @@ def days_in_month(self) -> int:
     @property
     def week_of_month(self) -> int:
         first_day_of_month = self.replace(day=1)
-
-        return self.week_of_year - first_day_of_month.week_of_year + 1
+        week_days = days_of_week()
+        first_day_index = week_days.index(first_day_of_month.day_of_week)
+        days_in_first_week = pendulum.DAYS_PER_WEEK - first_day_index
+        days_after_first_week = self.day - days_in_first_week
+        return math.ceil(days_after_first_week / pendulum.DAYS_PER_WEEK) + 1
```

Comment:
```suggestion
        return int(
            math.ceil((self.day + first_day_of_month.day_of_week - 1) / DAYS_PER_WEEK)
        )
```

This should give us what we want while taking care of the edge cases that needs fixing.